src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpChangeSignatureDialog.cs

```diff
@@ -0,0 +1,175 @@
+            ChangeSignatureDialog.VerifyOpen();
+            ChangeSignatureDialog.SelectParameter("int a");
+            ChangeSignatureDialog.ClickDownButton();
+            ChangeSignatureDialog.ClickOK();
+            ChangeSignatureDialog.VerifyClosed();
+
+            VerifyTextContains(@"
+class C
+{
+    public void Method(string b, int a) { }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ChangeSignature)]
+        public void VerifyRemoveParameter()
+        {
+            SetUpEditor(@"
+class C
+{
+    /// <summary>
+    /// A method.
+    /// </summary>
+    /// <param name=""a""></param>
+    /// <param name=""b""></param>
+    public void Method$$(int a, string b) { }
+
+    void Test()
+    {
+        Method(1, ""s"");
+    }
+}");
+
+            ChangeSignatureDialog.Invoke();
+            ChangeSignatureDialog.VerifyOpen();
+            ChangeSignatureDialog.SelectParameter("string b");
+            ChangeSignatureDialog.ClickUpButton();
+            ChangeSignatureDialog.ClickRemoveButton();
+            ChangeSignatureDialog.ClickOK();
+            ChangeSignatureDialog.VerifyClosed();
+
+            VerifyTextContains(@"
+class C
+{
+    /// <summary>
+    /// A method.
+    /// </summary>
+    /// <param name=""a""></param>
+    /// 
+    public void Method(int a) { }
+
+    void Test()
+    {
+        Method(1);
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.ChangeSignature)]
+        public void VerifyCrossLanguageGlobalUndo()
+        {
+            SetUpEditor(@"using VBProject;
+
+class Program
+{
+    static void Main(string[] args)
+    {
+        VBClass vb = new VBClass();
+        vb.Method$$(1, y: ""hello"");
+        vb.Method(2, ""world"");
+    }
+}");
+
+
+            VisualStudio.Instance.SolutionExplorer.AddProject("VBProject", WellKnownProjectTemplates.ClassLibrary, LanguageNames.VisualBasic);
+            Editor.SetText(@"
+Public Class VBClass
+    Public Sub Method(x As Integer, y As String)
+    End Sub
+End Class");
+
+            VisualStudio.Instance.SolutionExplorer.SaveAll();
+            VisualStudio.Instance.SolutionExplorer.AddProjectReference(ProjectName, "VBProject");
+            VisualStudio.Instance.SolutionExplorer.OpenFile(ProjectName, "Class1.cs");
+
+            WaitForAsyncOperations(FeatureAttribute.Workspace);
+
+            ChangeSignatureDialog.Invoke();
+            ChangeSignatureDialog.VerifyOpen();
+            ChangeSignatureDialog.SelectParameter("String y");
+            ChangeSignatureDialog.ClickUpButton();
+            ChangeSignatureDialog.ClickOK();
+            ChangeSignatureDialog.VerifyClosed();
+
+            VerifyTextContains(@"vb.Method(y: ""hello"", x: 1);");
+
+            VisualStudio.Instance.SolutionExplorer.OpenFile("VBProject", "Class1.vb");
+            VerifyTextContains(@"Public Sub Method(y As String, x As Integer)");
+
+            Editor.UndoLastGlobalAction();
```

Comment:
Well, "undo last local action" is just Edit.Undo, which has a keyboard shortcut (ctrl+z) and is what the customer actually does. UndoLastGlobalAction is a magical command with no default keyboard shortcut that we shouldn't be using here. Will change to just Editor.Undo().